slither/detectors/erc/erc20/arbitrary_send_erc20.py

```diff
@@ -28,66 +32,68 @@ def permit_results(self) -> List[Node]:
         return self._permit_results
 
     def _detect_arbitrary_from(self, contract: Contract):
-        for f in contract.functions:
-            all_high_level_calls = [
-                f_called[1].solidity_signature
-                for f_called in f.high_level_calls
-                if isinstance(f_called[1], Function)
-            ]
-            all_library_calls = [f_called[1].solidity_signature for f_called in f.library_calls]
-            if (
-                "transferFrom(address,address,uint256)" in all_high_level_calls
-                or "safeTransferFrom(address,address,address,uint256)" in all_library_calls
-            ):
-                if (
-                    "permit(address,address,uint256,uint256,uint8,bytes32,bytes32)"
-                    in all_high_level_calls
-                ):
-                    ArbitrarySendErc20._arbitrary_from(f.nodes, self._permit_results)
-                else:
-                    ArbitrarySendErc20._arbitrary_from(f.nodes, self._no_permit_results)
+        for func in contract.functions_entry_points:
+            has_permit = False
+            for node in func.nodes:
+                for ir in node.all_slithir_operations():
+                    # FIXME this is not always in the same order and produces causes permits to fall into non-permit
```

Comment:
There's a weird non-determinism issue where calls to `permit` are sometimes out-of-order and it changes which detector flags the test case that I need to fix.